cli/src/base.ts

```diff
@@ -75,4 +98,118 @@ export abstract class BaseCommand extends Command {
       return this.error(`Error accessing the API: ${err instanceof Error ? err.message : String(err)}`);
     }
   }
+
+  async getWorkspace(options: { allowCreate?: boolean } = {}) {
+    const xata = await this.getXataClient();
+    const workspaces = await xata.workspaces.getWorkspacesList();
+
+    if (workspaces.workspaces.length === 0) {
+      if (!options.allowCreate) {
+        return this.error('No workspaces found, please create one first');
+      }
+
+      const result = await prompts({
+        type: 'text',
+        name: 'name',
+        message: 'New workspace name'
+      });
+      const workspace = await xata.workspaces.createWorkspace({ name: result.name, slug: slugify(result.name) });
+      return workspace.id;
+    } else if (workspaces.workspaces.length === 1) {
+      const workspace = workspaces.workspaces[0].id;
+      this.log(`You only have a workspace, using it by default: ${workspace}"`);
+      return workspace;
+    }
+
+    const result = await prompts({
+      type: 'select',
+      name: 'workspace',
+      message: 'Select a workspace',
+      choices: workspaces.workspaces.map((workspace) => ({
+        title: workspace.name,
+        description: workspace.id,
+        value: workspace.id
+      }))
+    });
+
+    return String(result.workspace);
+  }
+
+  async getDatabase(workspace: string, options: { allowCreate?: boolean } = {}) {
+    const xata = await this.getXataClient();
+    const databases = await xata.databases.getDatabaseList(workspace);
+    const dbs = databases.databases || [];
+
+    if (dbs.length > 0) {
+      const choices = dbs.map((db) => ({
+        title: db.name,
+        value: db.name
+      }));
+
+      if (options.allowCreate) {
+        choices.splice(0, 0, { title: 'Create a new database', value: 'create' });
+      }
+
+      const result = await prompts({
+        type: 'select',
+        name: 'database',
+        message: dbs.length > 0 && options.allowCreate ? 'Select a database or create a new one' : 'Select a database',
```

Comment:
Same as before, instead of an allowCreate flag. Show the N databases + 1 option to create a new one